src/test/java/io/appform/dropwizard/actors/actor/metadata/MessageMetadataProviderTest.java

```diff
@@ -0,0 +1,108 @@
+package io.appform.dropwizard.actors.actor.metadata;
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Envelope;
+import io.appform.dropwizard.actors.actor.MessageMetadata;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+class MessageMetadataProviderTest {
+
+    private static final String CUSTOM_HEADER_KEY = "customHeaderKey";
+    private static final String CUSTOM_HEADER_VALUE = "value1234";
+
+    @Mock
+    private Envelope mockEnvelope;
+
+    @Mock
+    private AMQP.BasicProperties mockProperties;
+
+    private MessageMetadataProvider messageMetadataProvider;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+
+        List<String> messageMetaGeneratorClasses = Arrays.asList(CustomMessageMetaHeaderGenerator.class.getName());
+        messageMetadataProvider = new MessageMetadataProvider(messageMetaGeneratorClasses);
+    }
+
+    @Test
+    void shouldCreateBlankMessageMetadataWhenMessageMetaGeneratorClassesIsNull() {
+        MessageMetadataProvider messageMetadataProvider = new MessageMetadataProvider(null);
+        MessageMetadata messageMetadata = messageMetadataProvider.createMetadata(mockEnvelope, mockProperties);
+        assertNotNull(messageMetadata);
+
+        assertFalse(messageMetadata.isRedelivered());
+        assertEquals(0, messageMetadata.getDelayInMs());
+        assertFalse(messageMetadata.isExpired());
+
+        assertNull(messageMetadata.getContentType());
+        assertNull(messageMetadata.getContentEncoding());
+        assertNull(messageMetadata.getHeaders());
+        assertNull(messageMetadata.getDeliveryMode());
```

Comment:
Once boxed types are removed, fix these as well

Reply:
Will updated, once the other comment is resolved.